### PR TITLE
Adds account picture, account info, balance on DApps panel

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/22125

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Screenshot_20220406-121700_2

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java [hunk 1]
```diff
@@ -140,23 +163,100 @@ private void updateState() {
...
+    private void updateAccount() {
+        mBraveWalletPanelServices.getKeyringService().getSelectedAccount(CoinType.ETH, address -> {
+            String selectedAccount = "";
+            if (address != null && !address.isEmpty()) {
+                selectedAccount = address;
+                updateAccountInfo(selectedAccount);
+            } else {
+                if (mAccountsWithPermissions.size() != 0) {
+                    selectedAccount = mAccountsWithPermissions.iterator().next().address;
+                } else if (mAccountInfos.length != 0) {
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:184]: nit: can replace `!= 0` with `> 0`  or `!mAccountsWithPermissions.isEmpty()`

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java [hunk 2]
```diff
@@ -12,55 +12,83 @@
...
+    private HashSet<AccountInfo> mAccountsWithPermissions;
+    private ExecutorService mExecutor;
+    private Handler mHandler;
+    private BraveWalletPanelServices mBraveWalletPanelServices;
+
+    public interface BraveWalletPanelServices {
+        public AssetRatioService getAssetRatioService();
+        public BraveWalletService getBraveWalletService();
+        public KeyringService getKeyringService();
+        public JsonRpcService getJsonRpcService();
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:78]: can remove explicit `public` keywords here

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java [hunk 3]
```diff
@@ -140,23 +163,100 @@ private void updateState() {
...
+        if (selectedAccount.length == 0) {
+            return;
+        }
+        mBraveWalletPanelServices.getJsonRpcService().getChainId(CoinType.ETH, chainId -> {
+            SingleTokenBalanceHelper singleTokenBalanceHelper =
+                    new SingleTokenBalanceHelper(
+                            mBraveWalletPanelServices.getAssetRatioService(),
+                            mBraveWalletPanelServices.getJsonRpcService());
+            singleTokenBalanceHelper.getPerAccountBalances(chainId, "",
+                    "eth", 18, selectedAccount, () -> {
```
Comment (Pavneet-Sing) [inline android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:240]: nit: can move "eth" and "ETH" to WalletConstants if suitable for re-usability for API calls across wallet?

**Review by Pavneet-Sing — APPROVED** [review]
++

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java (hunk 1 above)
Comment (Pavneet-Sing) [reply android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:184]: not much, it just quickly makes it clear that it should be greater than one, as `!=0` can be positive or negative.

File: android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java (hunk 3 above)
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/crypto_wallet/modal/BraveWalletPanel.java:240]: I think what we will need to do when we work on multi chain support for DApps is: some kind of a helper class that will return a native token looking on a chainId. It had to be done in all places not only here. I would leave it for now.

**Pull request merged**